tests/test_training.py

```diff
@@ -96,3 +109,43 @@ def test_ipu_cli_training(self, load_type):
 
                 assert _IPU_AVAILABLE is True
                 self.call_cli_with_overrides("ipu", "16-true", load_type)
+
+    def test_ipu_mlp(self):
+
+        import torch
+        from torch import nn
+        import poptorch
+
+        dim = 32
+
+        class MLP(nn.Module):
+
+            def __init__(self):
+
+                super().__init__()
+
+                self.lin1 = nn.Linear(dim, dim)
+                self.lin2 = nn.Linear(dim, dim)
+
+            def forward(self, x, y):
+
+                out = self.lin1(x).maximum(torch.zeros(1))
+                out = self.lin2(out)
+
+                if self.training:
+                    loss = (out - y).pow(2)
+                    loss = poptorch.identity_loss(loss, reduction='sum')
+                    return out, loss
+
+                return out
+
+        x = torch.rand(dim)
+        y = torch.rand(dim)
+
+        mlp = MLP()
+        o1 = poptorch.Options()
+        o1.useIpuModel(True)
+        o2 = poptorch.optim.SGD(mlp.parameters(), lr=1e-4)
+        popmlp = poptorch.trainingModel(mlp, o1, o2)
+
+        z = popmlp(x, y)
```

Comment:
For reliability I think we want to patch the pop torch ipu availability here as well. 
I'm not sure if this test was run on its own on a machine without IPUs if it would fail. 
Or if the tests run in a different order - currently we don't have loads of parallelisation in it, but good to be safe. 
So maybe try something like the above suggestion?